src/Oscoin/Storage/Block/Memory.hs

```diff
@@ -57,6 +57,13 @@ mkBlockStore getBlocks modifyBlocks = (blockStoreReader, blockStoreWriter)
                 _ Seq.:|> gen  -> pure gen
         , lookupBlock = \h ->
             find (\b -> blockHash b == h) <$> getBlocks
+        , lookupBlockByHeight = \h ->
+            find (\b -> (blockHeight . blockHeader $ b) == h) <$> getBlocks
```

Comment:
For a blockchain to be valid its blocks must have height equaling their position in the chain. If we are sure that `Block.Memory` satisfies this invariant we could implement this by just requesting the block in the list at the given height. If not we should add a comment so that people can understand why we didn’t chose the simple version.

Reply:
I agree, but I think that here the `Seq` doesn't give us such guarantee (if `getBlocks` were using a `BlockChain c tx s` type, that would have been easier to enforce). As the memory implementation was always meant as a simplified store where performance is not the primary concern, I would be inclined to go with the simpler version. Will add a comment.

Reply:
d146be09